### PR TITLE
Update cppwinrt, Catch2, and Visual Studio CI

### PR DESCRIPTION
**Pull request by AtariDreams**
Update cppwinrt from 2.0.220608.4 to 2.0.220929.3, and Catch2 from 2.7.0 to 2.13.10, and finally, in order to compile cppwinrt, Visual Studio from 2019 to 2022

### CODE REVIEW

**Comment by AtariDreams** [comment]
This compiled locally for x86 and x64, but not for CI

**Comment by dunhor** [comment]
I wonder if it's running out of heap space? Though IIRC the error message for that is clearer than a generic ICE.

If that's the case, updating to use the amd64 host compiler might help fix it. It also might help updating from using VS 2019 build machines to VS 2022, if they are available. Of course, this assumes that this is the issue...

**Comment by AtariDreams** [comment]
> I wonder if it's running out of heap space? Though IIRC the error message for that is clearer than a generic ICE.
> 
> If that's the case, updating to use the amd64 host compiler might help fix it. It also might help updating from using VS 2019 build machines to VS 2022, if they are available. Of course, this assumes that this is the issue...

This seems to have worked. Thank you!

**Comment by AtariDreams** [comment]
@dunhor any updates?

File: tests/wiTest.cpp [hunk 1]
```diff
@@ -488,8 +488,8 @@ TEST_CASE("WindowsInternalTests::ResultMacros", "[result_macros]")
     REQUIRE_RETURNS_EXPECTED(E_hrNtAssertionFailure, [] { RETURN_IF_NTSTATUS_FAILED_EXPECTED(STATUS_ASSERTION_FAILURE); return S_OK; });
     REQUIRE_THROWS_RESULT(E_hrNtAssertionFailure, [] { THROW_IF_NTSTATUS_FAILED(STATUS_ASSERTION_FAILURE); });
     REQUIRE_THROWS_MSG(E_hrNtAssertionFailure, [] { THROW_IF_NTSTATUS_FAILED_MSG(STATUS_ASSERTION_FAILURE, "msg: %d", __LINE__); });
-    REQUIRE_LOG(E_hrNtAssertionFailure, [] { REQUIRE(STATUS_ASSERTION_FAILURE == LOG_IF_NTSTATUS_FAILED(STATUS_ASSERTION_FAILURE)); });
-    REQUIRE_LOG_MSG(E_hrNtAssertionFailure, [] { REQUIRE(STATUS_ASSERTION_FAILURE == LOG_IF_NTSTATUS_FAILED_MSG(STATUS_ASSERTION_FAILURE, "msg: %d", __LINE__)); });
+    REQUIRE_LOG(E_hrNtAssertionFailure, [] { REQUIRE(STATUS_ASSERTION_FAILURE == static_cast<DWORD>(LOG_IF_NTSTATUS_FAILED(STATUS_ASSERTION_FAILURE))); });
```
Comment (jonwis) [inline tests/wiTest.cpp:491]: Any idea why this is required? `STATUS_ASSERTION_FAILED` should be an NTSTATUS which is a LONG, and `LOG_IF_NTSTATUS_FAILED` should just be returning NTSTATUS itself.  Is it picking up some other definition of STATUS_ASSERTION_FAILED?

**Review by jonwis — APPROVED** [review]

File: tests/wiTest.cpp (hunk 1 above)
Comment (GameCubeGBA) [reply tests/wiTest.cpp:491]: Tests fail without it. It's a required warning when updating the library now.

Comment (AtariDreams) [reply tests/wiTest.cpp:491]: I guess stronger warnings this time, since VS2022 was updated. All I know is that the CI will fail without them.

**Review by dunhor — APPROVED** [review]

**Comment by AtariDreams** [comment]
Could someone please look at this?

**Pull request merged**